src/client/datascience/jupyter/kernelVariables.ts

```diff
@@ -418,6 +418,18 @@ export class KernelVariables implements IJupyterVariables {
                 }
             }
 
+            type SortableColumn = "name" | "type";
+            const sortColumn = request.sortColumn as SortableColumn;
+            const comparer = (a: IJupyterVariable, b: IJupyterVariable): number => {
+                if (!request.sortAscending) {
+                    return (a[sortColumn] < b[sortColumn]) ? 1 : -1;
+                } else {
+                    return (a[sortColumn] > b[sortColumn]) ? 1 : -1;
+                }
+            };
+            result.pageResponse.sort(comparer);
```

Comment:
This isn't going to work. It's only sorting a single page at a time. The page retrieved might even be the wrong one.

Imagine you had 10,000 variables, let's say with names like:

A1
A2
A3
.
.
B1
B2
B3
.

And you fetched the first page. It will return the A1 page. You then sort descending. It still fetch the first page, you'll just have that page sorted descending, but what the user wanted to see was the Z900 to Z800 set of variables.

Reply:
You have to push the sorting all the way into the python code.

Reply:
And that's rather complicated because it requires reworking how variables are fetched (we only fetch names and then with a separate request we fetch types).

Reply:
Ahh, I see where I went wrong. Thanks for all your input :) 
I shall try again